### ISSUE TITLE
Sharpness/silhouette heuristic false-positive tanks a genuinely sharp photo's score

### ISSUE THREAD

**Issue by thvd** [prasadabhishek/photo-quality-analyzer#7]
## Problem

A real, genuinely sharp photo scores `sharpness: 0.28` with the explanation `"High contrast edges found, but lacking internal texture (Silhouette?)"` -- the anti-silhouette heuristic mentioned in the v0.8.3 changelog ("Gradient Sparsity Check: Implemented volumetric texture analysis to distinguish paper-thin contrast edges from actual volumetric sharpness") appears to false-positive on this image.

## Repro

Attached (EXIF-stripped): ![repro](https://raw.githubusercontent.com/thvd/photo-quality-analyzer/mainline/bug-repro/img_5913_silhouette_bug_repro.jpg)

The foreground is a cut log stump with visible, crisp wood-grain detail, photographed in even outdoor daylight -- not a silhouette or flat/backlit subject. Visually it's clearly in sharp focus (the cut end shows distinct radial grain lines, bark texture is resolved, a small bird perched on top is also sharp).

```python
from photo_quality_analyzer_core.analyzer import evaluate_photo_quality
r = evaluate_photo_quality("img_5913_silhouette_bug_repro.jpg", enable_subject_detection=False, force_downsample=False)
print(r["metrics"]["sharpness"])
# {'score': 0.28064587074999997, 'explanation': 'High contrast edges found, but lacking internal texture (Silhouette?). ()'}
print(r["overallConfidence"], r["judgement"])
# 0.5169886362514247 Acceptable
```

Reproduces identically with `enable_subject_detection=True` (score is bit-for-bit the same), so this isn't a subject-detection/YOLO issue -- it's in the core global sharpness/silhouette-check path.

## Compounding issue (same photo pair, different failure)

With `enable_subject_detection=True`, YOLO26-nano also misclassifies the log stump as `"elephant (Conf 0.57)"`, which then triggers a focus veto (`"Best subject 'elephant (Conf 0.57)' is soft (0.08)"`) that further drags the score down -- a second, independent problem stacking on top of the silhouette false-positive.

## Impact

I'm using `evaluate_photo_quality` to automatically pick the best frame out of a camera burst (Immich stack primary selection). For this exact burst, a companion frame where the same subject is genuinely blurry/out of focus scored `sharpness: 1.0` via the "no subject detected -> use overall sharpness" fallback (since a busy, correctly-focused background dominated that frame's frequency content) -- so the tool picked the blurry frame as the winner over this sharp one. Two independent bugs pushing the result in opposite wrong directions on the same real-world photo pair.

Happy to share the full-resolution original privately if useful for debugging (this repro copy is downscaled and stripped of EXIF/GPS for privacy). Also happy to try a PR if you can point me at the relevant sharpness/silhouette-check function -- I found `_calculate_sharpness` in `analyzer.py` but wanted to check before digging into the heuristic's internals since I don't know the intended tuning target.

(Context: found via the same automation as #4, #5, #6.)
